aptos-move/framework/aptos-framework/sources/genesis.move

```diff
@@ -179,13 +190,47 @@ module aptos_framework::genesis {
             i = i + 1;
         };
 
+        // Mint initial balances that were not handled by validators
+        let i = 0;
+        let len = vector::length(&initial_balances);
+        while (i < len) {
+            let account = *vector::borrow(&initial_balance_accounts, i);
+            let initial_balance = *vector::borrow(&initial_balances, i);
+            // We have to create the account before minting
+            create_account(aptos_framework, account, initial_balance);
+            i = i + 1;
+        };
+
         // Destroy the aptos framework account's ability to mint coins now that we're done with setting up the initial
         // validators.
         aptos_coin::destroy_mint_cap(aptos_framework);
 
         stake::on_new_epoch();
     }
 
+    /// Creates an account allowing for the signer to be used elsewhere in genesis
+    fun create_account(aptos_framework: &signer, account: address, balance: u64): signer {
+        let signer = account::create_account(account);
+        coin::register<AptosCoin>(&signer);
+
+        if (balance > 0) {
+            aptos_coin::mint(aptos_framework, account, balance);
+        };
+        signer
+    }
+
+    /// Finds the balance of the account and removes it from the list
+    fun find_initial_balance(accounts: &mut vector<address>, balances: &mut vector<u64>, account: &address): u64 {
+        // Lookup account
+        let (found, index) = vector::index_of(accounts, account);
+        assert!(found, 404);
```

Comment:
probably want to use a clearer error here, we have error::not_found(EBALANCE) or something

Reply:
I'm pretty sure there isn't good error mapping in Genesis, but yes I can add that.